waspc/cli/src/Wasp/Cli/Command/Uninstall.hs

```diff
@@ -0,0 +1,70 @@
+module Wasp.Cli.Command.Uninstall
+  ( uninstall,
+  )
+where
+
+import Control.Exception (SomeException, try)
+import Control.Monad (when)
+import Control.Monad.Except (throwError)
+import Control.Monad.IO.Class (liftIO)
+import Data.Maybe (fromJust)
+import qualified StrongPath as SP
+import System.Directory
+  ( doesDirectoryExist,
+    doesFileExist,
+    getHomeDirectory,
+    removeDirectoryRecursive,
+    removeFile,
+  )
+import Wasp.Cli.Command (Command, CommandError (..))
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import qualified Wasp.Message as Msg
+
+-- Types
+data HomeDir
+
+-- | Removes Wasp CLI from the system.
+-- It removes the follwing:
+-- {home}/.local/share/wasp-lang
+-- {home}/.local/bin/wasp
+uninstall :: Command ()
+uninstall = do
+  cliSendMessageC $
+    Msg.Info "Uninstalling Wasp CLI ..."
+
+  uninstallResult <- liftIO (try removeWaspFiles :: IO (Either SomeException ()))
+  either
+    (throwError . CommandError "Unable to uninstall" . show)
+    pure
+    uninstallResult
+
+  cliSendMessageC $ Msg.Success "Uninstalled Wasp CLI"
+
+removeWaspFiles :: IO ()
+removeWaspFiles = do
```

Comment:
Depends if they are used somewhere else or just here! But ok, that is easy to move around.

Reply:
I would maybe use a bit different names: `let waspDirAbsPath` and `let waspBinAbsPath`. But no biggie, doesn't matter much.

Or I would even be tempted to do `(doesDirectoryExist . fromAbsDir)`, and `(removeDirectoryRecursive . fromAbsDir)`, and so on, since it is really property of these functions to define what kind of path they want to take in. If they were typed, we wouldn't have to do any conversion at all, but if we have to, we can do it right here where we call them.

I haven't been doing this in the past, but now I am getting to like it as an approach.